### PR TITLE
ansible_galaxy_install: add upgrade feature

### LINKED ISSUE #8351: ansible_galaxy_install has no update functionality that reports only updated collections as changed
### Summary

In module `ansible_galaxy_install` there is no update functionality, there is only `force: true` option that will report changed to all collections even if they were not updated. This probably also applies to roles and not only collections.

There should be a way to update collections and report only updated collections as `changed: true`. 
This was easy to accomplish using `ansible-galaxy` command and a simple `changed_when`.

In essence the module `ansible_galaxy_install` is less usable than the `ansible-galaxy` command.

### Issue Type

Feature Idea

### Component Name

ansible_galaxy_install

### Additional Information

```yaml (paste below)

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY

Add upgrade feature.

Fixes #8351 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

ansible_galaxy_install

### CODE REVIEW

File: plugins/modules/ansible_galaxy_install.py [hunk 1]
```diff
@@ -188,6 +202,7 @@ class AnsibleGalaxyInstall(ModuleHelper):
     output_params = ('type', 'name', 'dest', 'requirements_file', 'force', 'no_deps')
     module = dict(
         argument_spec=dict(
+            state=dict(type='str', choices=['present', 'latest'], default="present"),
```
Comment (felixfontein) [inline plugins/modules/ansible_galaxy_install.py:205]: Why use double quotes for the default, when using single quotes everywhere else in this line? :)

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

File: plugins/modules/ansible_galaxy_install.py (hunk 1 above)
Comment (russoz) [inline plugins/modules/ansible_galaxy_install.py:205]:
```suggestion
            state=dict(type='str', choices=['present', 'latest'], default='present'),
```

Comment (russoz) [reply plugins/modules/ansible_galaxy_install.py]: Done ;)

Comment (felixfontein) [reply plugins/modules/ansible_galaxy_install.py]: I don't care too much about which quotes you use, I just found it funny that you mixed them even in one line ;)

File: plugins/modules/ansible_galaxy_install.py [hunk 2]
```diff
@@ -32,6 +32,19 @@
...
 options:
+  state:
+    description:
+      - >
+        If O(state=present) then the collection or role will be installed.
+        Note that the collections and roles are not updated with this option.
+      - >
+        Currently the O(state=latest) is ignored unless O(type=collection), and it will
+        ensure the collection is installed and updated to the latest available version.
+      - Please note that using O(force=true) can be used to perform upgrade regardless of O(type).
```
Comment (russoz) [inline plugins/modules/ansible_galaxy_install.py:43]:
```suggestion
      - Please note that O(force=true) can be used to perform upgrade regardless of O(type).
```

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks for your contribution!

**Pull request merged**